Does not build on Mac.

Trying to build following instructions on the latest MacOS (Ventura 13.4.1) gives:

```
❯ make
make: clang++-10: No such file or directory
make: *** [build/.//Utils/Utils.o] Error 1
```

I have Xcode CLI tools installed and clang is version:

```
❯ clang++ -v
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

I guess it is just not supported. Would be cool to have it.

--- a/MnemonicShamirCLI/Makefile
+++ b/MnemonicShamirCLI/Makefile
@@ -1,7 +1,7 @@
 # Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
 # See the file LICENSE for licensing terms.
 
-CXX := clang++-10
+CXX := clang++
 CXX_FLAGS := -std=c++17 -g -fno-omit-frame-pointer -Wall
 LIB_INSTALL_DIR ?=
 USR_INCLUDE ?= $(LIB_INSTALL_DIR)/usr/local/include

